### ISSUE TITLE
Assignment Relationships

### ISSUE THREAD

**Issue by tarebyte** [education/classroom#25]
@johndbritton This is just an overview of the relationships as I view them for individual and group assignments.

Let me know what you think. (interaction diagram coming soon)

```
                               +----------------------------------+
+-------------------------+    | Organization                     |
| User                    |    |                                  |
|                         |    | has_many: individual_assignments |
| has_many :repo_accesses |    | has_many: group_assignments      |
| github_user_id          |    | has_many: repo_accesses          |
+-------------------------+    +----------------------------------+
                                                                   
                    +--------------------------+                   
                    | RepoAccess               |                   
                    |                          |                   
                    | belongs_to :organization |                   
                    | belongs_to :user         |                   
                    | github_team_id           |                   
                    +--------------------------+                   
                                                                   
+----------------------+                +-------------------------+
| AssignmentRepo       |                | GroupAssignmentRepo     |
|                      |                |                         |
| has_one :repo_access |                | has_many :repo_accesses |
| github_repo_id       |                | github_repo_id          |
+----------------------+                +-------------------------+
                                                                   
                               +----------------------------------+
+----------------------------+ | Grouping                         |
| IndividualAssignment       | |                                  |
|                            | | has_many :group_assignment_repos |
| has_many :assignment_repos | | title                            |
+----------------------------+ +----------------------------------+
                                                                   
                                              +-------------------+
                                              | GroupAssignment   |
                                              |                   |
                                              | has_one :grouping |
                                              +-------------------+

```

**Comment by johndbritton** [comment]
This makes sense to me, but shouldn't there be a relationship between `User` and `Grouping` or perhaps a model for `Group`?

I see that it would be possible to infer who is in which group based on the `RepoAccesses` associated with each `GroupAssignmentRepo` but how would the teacher reuse the `Grouping` when creating the next `GroupAssignment`?

**Comment by tarebyte** [comment]
@johndbritton nice catch! I gave the `Organization` `Groupings` which is now `has_many :groups`. That way a teacher can select from a `Grouping` if they don't want the teams to change.
 
```
                               +----------------------------------+
+-------------------------+    | Organization                     |
| User                    |    |                                  |
|                         |    | has_many :individual_assignments |
| has_many :repo_accesses |    | has_many :group_assignments      |
| github_user_id          |    | has_many :repo_accesses          |
+-------------------------+    | has_many :groupings              |
                               +----------------------------------+
                    +--------------------------+                   
                    | RepoAccess               |                   
                    |                          |                   
                    | belongs_to :organization |                   
                    | belongs_to :user         |                   
                    | github_team_id           |                   
                    +--------------------------+                   
                                                                   
+----------------------+                +-------------------------+
| AssignmentRepo       |                | Group                   |
|                      |                |                         |
| has_one :repo_access |                | has_many :repo_accesses |
| github_repo_id       |                | belongs_to :grouping    |
+----------------------+                +-------------------------+
                                                                   
                                               +------------------+
+----------------------------+                 | Grouping         |
| IndividualAssignment       |                 |                  |
|                            |                 | has_many :groups |
| has_many :assignment_repos |                 | title            |
+----------------------------+                 +------------------+
                                                                   
                                            +---------------------+
                                            | GroupAssignmentRepo |
                                            |                     |
                                            | has_one :group      |
                                            | github_repo_id      |
                                            +---------------------+
                                                                   
                               +----------------------------------+
                               | GroupAssignment                  |
                               |                                  |
                               | has_one :grouping                |
                               | has_many :group_assignment_repos |
                               +----------------------------------+

```

**Comment by tarebyte** [comment]
Also I'm going to update `AssignmentRepo` to `IndividualAssignmentRepo`, just fits the naming convention better.

**Comment by tarebyte** [comment]
Implemented this in #26
